packages/core/src/core/rsbuild.ts

```diff
@@ -19,6 +19,47 @@ const isMultiCompiler = <
   return 'compilers' in compiler && Array.isArray(compiler.compilers);
 };
 
+const autoExternalNodeModules: (
+  data: Rspack.ExternalItemFunctionData,
+  callback: (
+    err?: Error,
+    result?: Rspack.ExternalItemValue,
+    type?: Rspack.ExternalsType,
+  ) => void,
+) => void = ({ context, request, dependencyType, getResolve }, callback) => {
+  if (!request || request.startsWith('node:')) {
+    return callback();
+  }
+
+  const doExternal = () => {
+    callback(
+      undefined,
+      `${dependencyType === 'commonjs' ? 'commonjs' : 'module-import'} ${request}`,
+    );
+  };
+  if (/node_modules/.test(request)) {
+    return doExternal();
+  }
+
+  const resolver = getResolve?.();
+
+  if (!resolver) {
+    return callback();
+  }
+
+  resolver(context!, request!, (err, resolvePath) => {
+    if (err) {
+      // ignore resolve error
+      return callback();
+    }
+
+    if (resolvePath && /node_modules/.test(resolvePath!)) {
```

Comment:
We could put `/node_modules/.test` logic into `doExternal` to convergence the logic.

Reply:
Never mind, I just saw two same logic to test `node_modules`.